// @tiptap/extension-font-family/dist/tiptap-extension-font-family.umd.js
(function (global, factory) {
  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@tiptap/core'), require('@tiptap/extension-text-style')) :
  typeof define === 'function' && define.amd ? define(['exports', '@tiptap/core', '@tiptap/extension-text-style'], factory) :
  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['@tiptap/extension-font-family'] = {}, global.core));
}(this, (function (exports, core) { 'use strict';

  const FontFamily = core.Extension.create({
      name: 'fontFamily',
      defaultOptions: {
          types: ['textStyle'],
      },
      addGlobalAttributes() {
          return [
              {
                  types: this.options.types,
                  attributes: {
                      fontFamily: {
                          default: null,
                          renderHTML: attributes => {
                              if (!attributes.fontFamily) {
                                  return {};
                              }
                              return {
                                  style: `font-family: ${attributes.fontFamily}`,
                              };
                          },
                          parseHTML: element => ({
                              fontFamily: element.style.fontFamily.replace(/['"]+/g, ''),
                          }),
                      },
                  },
              },
          ];
      },
      addCommands() {
          return {
              setFontFamily: fontFamily => ({ chain }) => {
                  return chain()
                      .setMark('textStyle', { fontFamily })
                      .run();
              },
              unsetFontFamily: () => ({ chain }) => {
                  return chain()
                      .setMark('textStyle', { fontFamily: null })
                      .removeEmptyTextStyle()
                      .run();
              },
          };
      },
  });

  exports.FontFamily = FontFamily;
  exports.default = FontFamily;

  Object.defineProperty(exports, '__esModule', { value: true });

})));
